Задача "Подпроекты"

Artur0209 opened the pull request:
1)Реализовал методы по созданию,обновлению,поиск по фильтрации(название и статус проекта), получение проекта по id. 2)Написал валидацию сервиса и контроллера.
3)Написал фильтры проекта по названию и статусу

ArturBerd committed:
Задача "Подпроекты"
1)Реализовал методы по созданию,обновлению,поиск по фильтрации(название и статус проекта), получение проекта по id.
2)Написал валидацию сервиса и контроллера.
3)Написал фильтры проекта по названию и статусу

ArturBerd committed:
Задача "Подпроекты"
1)Реализовал методы по созданию,обновлению,поиск по фильтрации(название и статус проекта), получение проекта по id.
2)Написал валидацию сервиса и контроллера.
3)Написал фильтры проекта по названию и статусу

DmShevchuk commented on src/main/java/faang/school/projectservice/controller/SubProjectController.java:
```diff
@@ -0,0 +1,58 @@
+    private final ProjectService projectService;
+    private final ValidatorSubProjectController validatorSubProjectController;
+
+    @PostMapping("/create")
```
При наименовании эндпоинтов не используются глаголы.
В данном случае можно просто оставить @PostMapping

DmShevchuk commented on src/main/java/faang/school/projectservice/controller/SubProjectController.java:
```diff
@@ -0,0 +1,58 @@
+        return projectService.createSubProject(createSubProjectDto);
+    }
+
+    @PutMapping("/update/{id}")
```
Аналогично комментарию выше + id запихнуть в DTO и оставить только PutMapping

DmShevchuk commented on src/main/java/faang/school/projectservice/controller/SubProjectController.java:
```diff
@@ -0,0 +1,58 @@
+        return projectService.updateSubProject(id, createSubProjectDto);
+    }
+
+    @GetMapping("/filter")
```
Оставить просто GetMapping

DmShevchuk commented on src/main/java/faang/school/projectservice/dto/client/subprojectdto/CreateSubProjectDto.java:
```diff
@@ -0,0 +1,16 @@
+
+@AllArgsConstructor
+@Data
+public class CreateSubProjectDto {
```
В проекте ДТО именуем по шаблону: <ИМЯ СУЩНОСТИ><ОПЕРАЦИЯ>Dto

В данном случае будет SubProjectCreateDto

DmShevchuk commented on src/main/java/faang/school/projectservice/service/ProjectService.java:
```diff
@@ -0,0 +1,89 @@
+        validatorProjectService.validateProjectExistence(createSubProjectDto.getParentProjectId());
+
+        Project parentProject = projectRepository.findById(createSubProjectDto.getParentProjectId())
+                .orElseThrow(() -> new EntityNotFoundException("Родительский проект не найден"));
```
+ к сообщению добавить указание id, по которому не нашли объект

DmShevchuk commented on src/main/java/faang/school/projectservice/service/ProjectService.java:
```diff
@@ -0,0 +1,89 @@
+        validatorProjectService.validateVisibility(parentProject, createSubProjectDto.getVisibility());
+
+        Project projectToUpdate = projectRepository.findById(projectId)
+                .orElseThrow(() -> new EntityNotFoundException("Проект не найден"));
```
к сообщению добавить указание id, по которому не нашли объект

DmShevchuk commented on src/main/java/faang/school/projectservice/service/ProjectService.java:
```diff
@@ -0,0 +1,89 @@
+
+    public Project getProjectById(Long projectId) {
+        return projectRepository.findById(projectId)
+                .orElseThrow(() -> new EntityNotFoundException("Проект не найден"));
```
к сообщению добавить указание id, по которому не нашли объект

ArturBerd committed:
Задача "Подпроекты"
1)Подправил код:добавил в дто id,оставил анотации @PostMapping @GetMapping..., исправил название дто согласно шаблону
